src/elements/content-sidebar/withSidebarAnnotations.js

```diff
@@ -0,0 +1,179 @@
+            const fileVersionId = getProp(match, 'params.fileVersionId');
+
+            if (fileVersionId && fileVersionId !== currentFileVersionId) {
+                history.replace(getAnnotationsPath(currentFileVersionId, annotationId));
+            }
+        };
+
+        componentDidUpdate(prevProps: Props) {
+            const { annotatorState, getAnnotationsMatchPath, location }: Props = this.props;
+            const { annotatorState: prevAnnotatorState, location: prevLocation }: Props = prevProps;
+            const { activeAnnotationId, annotation } = annotatorState;
+            const { activeAnnotationId: prevActiveAnnotationId, annotation: prevAnnotation } = prevAnnotatorState;
+
+            const match = getAnnotationsMatchPath(location);
+            const prevMatch = getAnnotationsMatchPath(prevLocation);
+            const fileVersionId = getProp(match, 'params.fileVersionId');
+            const hasActiveAnnotationChanged = prevActiveAnnotationId !== activeAnnotationId;
+            const isAnnotationsPath = !!match;
+            const isTransitioningToAnnotationPath = activeAnnotationId && !isAnnotationsPath;
+            const prevFileVersionId = getProp(prevMatch, 'params.fileVersionId');
+
+            if (annotation && prevAnnotation !== annotation) {
+                this.addAnnotation();
+            }
+
+            // Active annotation id changed. If location is currently an annotation path or
+            // if location is not currently an annotation path but the active annotation id
+            // transitioned from falsy to truthy, update the location accordingly
+            if (hasActiveAnnotationChanged && (isAnnotationsPath || isTransitioningToAnnotationPath)) {
+                this.updateActiveAnnotation();
+            }
+
+            if (prevFileVersionId !== fileVersionId) {
+                this.updateActiveVersion();
+            }
+        }
+
+        addAnnotation() {
+            const {
+                annotatorState: { action, annotation, meta },
+                api,
+                currentUser,
+                file,
+                fileId,
+                getAnnotationsMatchPath,
+                isOpen,
+                location,
+            } = this.props;
+            const { requestId } = meta || {};
+            const isAnnotationsPath = getAnnotationsMatchPath(location);
+            const isPending = action === 'create_start';
+
+            // TODO: need to address in follow on -- currentUser may be undefined here but is never fetched for sure until ActivitySidebar
+            if (!currentUser) {
+                throw getBadUserError();
+            }
+
+            const feedAPI = api.getFeedAPI(false);
+            const { items: hasItems } = feedAPI.getCachedItems(fileId) || {};
+
+            // If there are existing items in the cache for this file, then patch the cache with the new annotation
+            if (hasItems) {
```

Comment:
Is this check needed? It looks like we do a check within `updateFeedItem`.

Reply:
I'll add to the comment in the code, but yeah it's needed because without the check, a fresh load of the `ActivitySidebar` will check the cache and see that there is already an entry for the feed items, in which case it will render that single annotation before refreshing with all the activity items